staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_discovery.go

```diff
@@ -20,15 +20,17 @@ import (
 	"net/http"
 	"strings"
 	"sync"
+	"sync/atomic"
 
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apiserver/pkg/endpoints/discovery"
 )
 
+type versionDiscoveryMap map[schema.GroupVersion]*discovery.APIVersionHandler
+
 type versionDiscoveryHandler struct {
-	// TODO, writing is infrequent, optimize this
-	discoveryLock sync.RWMutex
-	discovery     map[schema.GroupVersion]*discovery.APIVersionHandler
+	discoveryLock    sync.RWMutex
```

Comment:
Is that lock still needed?

Reply:
Yes, the lock is used for the writers (but not the readers).

https://golang.org/pkg/sync/atomic/#Value in the "Example (ReadMostly)"